distsql/request_builder.go

```diff
@@ -290,6 +291,7 @@ func (builder *RequestBuilder) SetFromSessionVars(sv *variable.SessionVars) *Req
 	builder.RequestSource.RequestSourceInternal = sv.InRestrictedSQL
 	builder.RequestSource.RequestSourceType = sv.RequestSourceType
 	builder.StoreBatchSize = sv.StoreBatchSize
+	builder.Request.ResourceGroupName = sv.ResourceGroupName
```

Comment:
Please check the global flag @@global.tidb_enable_resource_control here.

Reply:
After discussing, we are going to set default resource group name in initializing session context, so no need to check here.